### PR TITLE
Fix extension contribution types

### LINKED ISSUE #313: registerExtension/IExtensionManifest Object literal may only specify known properties
When attempting to call registerExtension with the contents of an extension's package.json, certain Json elements do not exist on IExtensionManifest/IRelaxedExtensionManifest

Attempting to call registerExtension with the content from [vscode-mock-debug's package.json](https://github.com/microsoft/vscode-mock-debug/blob/main/package.json) we get the following errors:

```
ERROR(vue-tsc)  No overload matches this call.
  The last overload gave the following error.
    Object literal may only specify known properties, and 'type' does not exist in type '{ url: string; }'.
      Object literal may only specify known properties, and 'enablement' does not exist in type 'ICommand'.
        Object literal may only specify known properties, and 'enablement' does not exist in type 'ICommand'.
          Object literal may only specify known properties, and 'languages' does not exist in type 'IDebugger'.
 FILE  C:/Users/dkattan/source/repos/immybot-ref/frontend/src/views/ScriptEditorV2/src/features/debugger.ts:767:9

    765 |       private: true,
    766 |       repository: {
  > 767 |         type: "git",
        |         ^^^^
    768 |         url: "https://github.com/Microsoft/vscode-mock-debug.git",
    769 |       },
    770 |       bugs: {

 ERROR(vue-tsc)  No overload matches this call.
  The last overload gave the following error.
    Object literal may only specify known properties, and 'type' does not exist in type '{ url: string; }'.
      Object literal may only specify known properties, and 'enablement' does not exist in type 'ICommand'.
        Object literal may only specify known properties, and 'enablement' does not exist in type 'ICommand'.
          Object literal may only specify known properties, and 'languages' does not exist in type 'IDebugger'.
 FILE  C:/Users/dkattan/source/repos/immybot-ref/frontend/src/views/ScriptEditorV2/src/features/debugger.ts:860:13

    858 |             title: "Debug File",
    859 |             category: "Mock Debug",
  > 860 |             enablement: "!inDebugMode",
        |             ^^^^^^^^^^
    861 |             icon: "$(debug-alt)",
    862 |           },
    863 |           {

 ERROR(vue-tsc)  No overload matches this call.
  The last overload gave the following error.
    Object literal may only specify known properties, and 'type' does not exist in type '{ url: string; }'.
      Object literal may only specify known properties, and 'enablement' does not exist in type 'ICommand'.
        Object literal may only specify known properties, and 'enablement' does not exist in type 'ICommand'.
          Object literal may only specify known properties, and 'languages' does not exist in type 'IDebugger'.
 FILE  C:/Users/dkattan/source/repos/immybot-ref/frontend/src/views/ScriptEditorV2/src/features/debugger.ts:867:13

    865 |             title: "Run File",
    866 |             category: "Mock Debug",
  > 867 |             enablement: "!inDebugMode",
        |             ^^^^^^^^^^
    868 |             icon: "$(play)",
    869 |           },
    870 |           {
```

I was able to track down command.enablement in the vscode source: https://github.com/microsoft/vscode/blob/2434b7da03ff51e5a56c4f4a4674167e49470419/src/vs/workbench/services/actions/common/menusExtensionPoint.ts#L577-L584

But I'm not sure how we'd be able to include these types since they are nested in IExtensionManifest which appears to be coming directly from vscode.

I suppose I can work around it by suppressing the error messages or doing JSON.parse(JSON.stringify()) similar to what you are doing in rollup-vsix-plugin.ts

Thoughts?

### PR DESCRIPTION
**Pull request by CGNonofr**
IExtensionManifest type from vscode is wrong, i've fixed them in https://github.com/microsoft/vscode/pull/202360 and I've included the change in the patch here

also include Task service override (sorry @kaisalmen I was wrong, there were remaining ones :p)

fix #313

### CODE REVIEW

File: scripts/vscode.patch [hunk 1]
```diff
@@ -647,6 +647,1318 @@ index fdefc170db0..1d3bc4affc5 100644
...
  
+diff --git a/src/vs/platform/extensions/common/extensions.ts b/src/vs/platform/extensions/common/extensions.ts
+index 222f6b91861..4ccb180248d 100644
+--- a/src/vs/platform/extensions/common/extensions.ts
++++ b/src/vs/platform/extensions/common/extensions.ts
+@@ -3,6 +3,7 @@
+  *  Licensed under the MIT License. See License.txt in the project root for license information.
+  *--------------------------------------------------------------------------------------------*/
+ 
++import { IJSONSchema } from 'vs/base/common/jsonSchema';
```
Comment (dkattan) [inline scripts/vscode.patch:658]: Good find!

**Comment by kaisalmen** [comment]
> IExtensionManifest type from vscode is wrong, i've fixed them in https://github.com/microsoft/vscode/pull/202360 and I've included the change in the patch here

Nice, did you see that the build pipeline failed in the PR?

**Comment by kaisalmen** [comment]
@CGNonofr Sie note: Are you aware the sourcemaps are broken?
image

They are looking for `<package>./src/...` code. Is this fixable /  a fix desirec?

**Review by kaisalmen — APPROVED** [review]
LGTM

**Comment by kaisalmen** [comment]
@CGNonofr will you release `1.85.2` once this is merged?

**Comment by CGNonofr** [comment]
> > IExtensionManifest type from vscode is wrong, i've fixed them in [microsoft/vscode#202360](https://github.com/microsoft/vscode/pull/202360) and I've included the change in the patch here
> 
> Nice, did you see that the build pipeline failed in the PR?

Should be fine now

> @CGNonofr Sie note: Are you aware the sourcemaps are broken? image
> 
> They are looking for `<package>./src/...` code. Is this fixable / a fix desired?

Yep, sourcemaps for this extension are broken on the VSCode side (they reference the sources while they are not bundled)

Vite should not even try to analyse those files though but I didn't find any way to prevent it from doing it

> @CGNonofr will you release `1.85.2` once this is merged?

Sure

**Pull request merged**